pyensign/ensign.py

```diff
@@ -199,12 +209,13 @@ async def subscribe(self, *topics, query="", consumer_group=None):
             # Get the ID of the topic
             topic_ids.append(str(self._resolve_topic(topic)))
 
-        async for event in self.client.subscribe(
+        # Run the subscriber
+        await self.client.subscribe(
```

Comment:
Note: This will break the examples

Reply:
Below is what our "minimal" Python example looks like now. Can you show how it will be different when we do the next merge to main?

```python
import json
import asyncio
from datetime import datetime

from pyensign.ensign import Ensign
from pyensign.events import Event

TOPIC = "chocolate-covered-espresso-beans"

async def publish(client, topic, events):
    await asyncio.sleep(1)
    errors = await client.publish(topic, events)
    if errors:
        print("Failed to publish events: {}".format(errors))
    return errors

async def subscribe(client, topic):
    id = await(client.topic_id(topic))
    async for event in client.subscribe(id):
        msg = json.loads(event.data)
        print("At {}, {} sent you the following message: {}".format(msg["timestamp"], msg["sender"], msg["message"]))
        return event

async def main():
    # Create an Ensign client
    client = Ensign(
        # endpoint="staging.ensign.world:443", # uncomment if in staging
        # auth_url="https://auth.ensign.world" # uncomment if in staging
    )

    # Create topic if it doesn't exist
    if not await client.topic_exists(TOPIC):
        await client.create_topic(TOPIC)

    # Create an event with some data
    msg = {
        "sender": "Enson the Sea Otter",
        "timestamp": datetime.now().isoformat(),
        "message": "You're looking smart today!",
    }
    data = json.dumps(msg).encode("utf-8")
    event = Event(data, mimetype="application/json")

    # Create the publisher and subscriber tasks
    pub = publish(client, TOPIC, event)
    sub = subscribe(client, TOPIC)

    # Wait for the tasks to complete
    await asyncio.gather(pub, sub)

if __name__ == "__main__":
    asyncio.run(main())
```

Reply:
Something like this (untested).

```python
received = asyncio.Event()

async def print_event(event):
    msg = json.loads(event.data)
    print("At {}, {} sent you the following message: {}".format(msg["timestamp"], msg["sender"], msg["message"]))
    received.set()

async def main():
    # Create an Ensign client
    client = Ensign(
        # endpoint="staging.ensign.world:443", # uncomment if in staging
        # auth_url="https://auth.ensign.world" # uncomment if in staging
    )

    # Create topic if it doesn't exist
    client.ensure_topic_exists(TOPIC)

    # Create an event with some data
    msg = {
        "sender": "Enson the Sea Otter",
        "timestamp": datetime.now().isoformat(),
        "message": "You're looking smart today!",
    }
    data = json.dumps(msg).encode("utf-8")
    event = Event(data, mimetype="application/json")

    # Publish and subscribe
    await client.subscribe(TOPIC, on_event=print_event)
    await asyncio.sleep(1)
    await client.publish(client, TOPIC, event)
    await received.wait()

if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(main())
```